crates/bevy_ecs/src/world/mod.rs

```diff
@@ -2184,6 +2184,53 @@ impl World {
                 .get_mut_by_id(component_id)
         }
     }
+
+    /// Iterates over all resources in the world
+    pub fn iter_resources(&self) -> impl Iterator<Item = (&ComponentInfo, Ptr<'_>)> {
+        self.storages.resources.iter().map(|(component_id, data)| {
+            let component_info = self.components.get_info(component_id).unwrap_or_else(|| {
+                panic!("ComponentInfo should exist for all resources in the world, but it does not for {:?}", component_id);
+            });
+            let ptr = data.get_data().unwrap_or_else(|| {
+                panic!(
+                    "When iterating all resources, resource of type {} was supposed to exist, but did not.",
+                    component_info.name()
+                )
+            });
+            (component_info, ptr)
+        })
+    }
+
+    /// Mutably iterates over all resources in the world
+    pub fn iter_resources_mut(&mut self) -> impl Iterator<Item = (&ComponentInfo, MutUntyped<'_>)> {
+        self.storages.resources.iter().map(|(component_id, data)| {
+            let component_info = self.components.get_info(component_id).unwrap_or_else(|| {
+                panic!("ComponentInfo should exist for all resources in the world, but it does not for {:?}", component_id);
+            });
+
+            let (ptr, ticks) = data.get_with_ticks().unwrap_or_else(|| {
+                panic!(
+                    "When iterating all resources, resource of type {} was supposed to exist, but did not.",
+                    component_info.name()
+                )
+            });
+
+            // SAFETY:
+            // - ???
```

Comment:
I'm not super confident how to verify the safety of this. The method docs say: “This should never alias the underlying ticks. All access must be unique”. What would aliasing the underlying ticks look like?

Reply:
Aliasing the underlying ticks would require you to hold references to the ticks multiples times, since you only ever take reference to one at a time you should be safe.

This is more of a concern in systems where you can take access to multiple resources simultaneously i.e. ensure two of those accesses aren't for the same resource.